tests/unittest/llmapi/test_deepseek_v4_tokenizer.py

```
@@ -0,0 +1,449 @@
+        processor=None,
+        conversation=[
+            {
+                "role": "user",
+                "content": "hello",
+            }
+        ],
+        add_generation_prompt=True,
+        mm_placeholder_counts=[{}],
+    )
+
+    assert prompt == ("<｜begin▁of▁sentence｜><｜User｜>hello<｜Assistant｜></think>")
```

Ruff (0.15.18)

[warning] String contains ambiguous `｜` (FULLWIDTH VERTICAL LINE). Did you mean `|` (VERTICAL LINE)?

(RUF001)